orion/primitives/tadgan.py

```diff
@@ -82,244 +142,317 @@ def __getstate__(self):
-
-                for j in range(self.iterations_critic):
-                    x = minibatch[j * self.batch_size: (j + 1) * self.batch_size]
-                    y = y_minibatch[j * self.batch_size: (j + 1) * self.batch_size]
-                    z = np.random.normal(size=(self.batch_size, self.latent_dim, 1))
-                    epoch_cx_loss.append(
-                        self.critic_x_model.train_on_batch([y, z], [valid, fake, delta]))
-                    epoch_cz_loss.append(
-                        self.critic_z_model.train_on_batch([x, z], [valid, fake, delta]))
-
-                epoch_g_loss.append(
-                    self.encoder_generator_model.train_on_batch([x, z], [valid, valid, y]))
-
-            cx_loss = np.mean(np.array(epoch_cx_loss), axis=0)
-            cz_loss = np.mean(np.array(epoch_cz_loss), axis=0)
-            g_loss = np.mean(np.array(epoch_g_loss), axis=0)
-            print('Epoch: {}/{}, [Dx loss: {}] [Dz loss: {}] [G loss: {}]'.format(
-                epoch, self.epochs, cx_loss, cz_loss, g_loss))
-
-    def fit(self, X, y=None, **kwargs):
-        """Fit the TadGAN.
+        cx_real = self.critic_x(y)
+        cx_fake = self.critic_x(x_)
+        cx_real_loss = self._wasserstein_loss(real, cx_real)
+        cx_fake_loss = self._wasserstein_loss(fake, cx_fake)
+        cx_gp = self._gradient_penalty_loss(y, x_, self.critic_x)
+        cx_loss = cx_real_loss + cx_fake_loss + 10 * cx_gp
+
+        # Critic z loss
+        z_ = self.encoder(X)
+        cz_real = self.critic_z(z)
+        cz_fake = self.critic_z(z_)
+        cz_real_loss = self._wasserstein_loss(real, cz_real)
+        cz_fake_loss = self._wasserstein_loss(fake, cz_fake)
+        cz_gp = self._gradient_penalty_loss(z, z_, self.critic_z)
+        cz_loss = cz_real_loss + cz_fake_loss + 10 * cz_gp
+
+        # Encoder-Generator Loss
+        x_rec_ = self.generator(z_)
+        eg_cx_fake_loss = self._wasserstein_loss(real, cx_fake)
+        eg_cz_fake_loss = self._wasserstein_loss(real, cz_fake)
+        eg_mse = MeanSquaredError()(y, x_rec_)
+        eg_loss = eg_cx_fake_loss + eg_cz_fake_loss + 10 * eg_mse
+
+        batch_loss = [
+            [cx_loss, cx_real_loss, cx_fake_loss, cx_gp],
+            [cz_loss, cz_real_loss, cz_fake_loss, cz_gp],
+            [eg_loss, eg_cx_fake_loss, eg_cz_fake_loss, eg_mse]
+        ]
+        output = self._format_losses(batch_loss)
+
+        return output
+
+    def fit(self, X: ndarray, y: Optional[ndarray] = None, **kwargs) -> None:
+        """Fit the TadGAN model.
 
         Args:
             X (ndarray):
-                N-dimensional array containing the input training sequences for the model.
+                N-dimensional array containing the input encoder sequences.
             y (ndarray):
-                N-dimensional array containing the target sequences we want to reconstruct.
+                Optional. N-dimensional array containing the input critic x sequences.
```

Comment:
can you verify these docstrings? for `y` I think it's difficult for the user to understand what is critic x. I would rather keep it as before `target sequences the model is reconstructing`.

Reply:
Reverted